mypy/checker.py

```diff
@@ -742,7 +742,10 @@ def is_implicit_any(t: Type) -> bool:
                 for arg in item.arguments:
                     init = arg.initialization_statement
                     if init:
-                        self.accept(init)
+                        fmt = 'Incompatible default initializer for argument "{}"'
```

Comment:
To shorten the message a bit, I'd suggest leaving out "initializer" here and replacing that word with "default" three lines below.